### PR TITLE
MINOR: Protocol schema refactor follow-up

### PR DESCRIPTION
**Pull request by ijuma**
- Use constants in a few places that were missed
- Remove ProtoUtils by moving its methods to Schema
- Merge SchemaVisitor and SchemaVisitorAdapter
- Change SchemaVisitor package.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @hachikuji

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java [hunk 1]
```diff
@@ -326,16 +327,15 @@ public static void main(String[] args) {
 
     private static boolean retainsBufferReference(Schema schema) {
         final AtomicReference<Boolean> foundBufferReference = new AtomicReference<>(Boolean.FALSE);
-        SchemaVisitor detector = new SchemaVisitorAdapter() {
+        SchemaVisitor detector = new SchemaVisitor() {
             @Override
             public void visit(Type field) {
-                if (field == BYTES || field == NULLABLE_BYTES || field == RECORDS) {
+                if (field == BYTES || field == NULLABLE_BYTES || field == RECORDS)
                     foundBufferReference.set(Boolean.TRUE);
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: nit: could we just use `true`?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: Yes, definitely.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java [hunk 2]
```diff
@@ -326,16 +327,15 @@ public static void main(String[] args) {
...
-                }
             }
         };
         foundBufferReference.set(Boolean.FALSE);
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: This seems redundant.

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: Yeah, simplified it all in my latest commit.

**Review by hachikuji — APPROVED** [review]
LGTM other than a couple nits.

File: clients/src/main/java/org/apache/kafka/common/protocol/types/SchemaVisitor.java [hunk 3]
```diff
@@ -14,14 +14,13 @@
...
-import org.apache.kafka.common.protocol.types.Type;
-
-public interface SchemaVisitor {
-    void visit(Schema schema);
-    void visit(ArrayOf array);
-    void visit(Type field);
+/**
+ * Override one or more of the visit methods with the desired logic.
+ */
+public abstract class SchemaVisitor {
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/common/protocol/types/SchemaVisitor.java]: I guess we could also move this into `Schema` and just call it `Visitor`?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/protocol/types/SchemaVisitor.java]: Yeah, sounds good.

**Comment by hachikuji** [comment]
The updates LGTM.

**Comment by hachikuji** [comment]
Merging to trunk.

**Pull request closed without merging**